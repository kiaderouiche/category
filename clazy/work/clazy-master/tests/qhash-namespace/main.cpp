#include <QtCore/QString>
QT_BEGIN_NAMESPACE
struct A { };
struct E { };
uint qHash(A) { return 0; }; // OK
namespace NS {

    struct B { struct B2 {}; struct B3 {}; };
    uint qHash(B) { return 0; }; // OK
    uint qHash(B::B2) { return 0; }; // OK
    namespace NS2 {
        struct C {};
        uint qHash(C) { return 0; }; // OK
    }
    uint qHash(NS2::C) { return 0; }; // Warn
    uint qHash(::A) { return 0; } // Warn
    uint qHash(B::B3) { return 0; }; // OK
    enum class EnumClass
    {
        One
    };
    uint qHash(EnumClass) { return 0; } // OK
}
uint qHash(NS::B) { return 0; } // Warn
uint qHash(NS::B *) { return 0; } // Warn
uint qHash(NS::B::B3) { return 0; }; // Warn


QT_END_NAMESPACE
uint qHash(E) { return 0; } // Warn
