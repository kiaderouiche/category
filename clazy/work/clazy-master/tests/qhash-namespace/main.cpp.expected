qhash-namespace/main.cpp:15:5: warning: Move qHash(NS::NS2::C) to NS::NS2 namespace for ADL lookup [-Wclazy-qhash-namespace]
qhash-namespace/main.cpp:16:5: warning: Move qHash(A) out of namespace NS [-Wclazy-qhash-namespace]
qhash-namespace/main.cpp:24:1: warning: Move qHash(NS::B) into NS namespace for ADL lookup [-Wclazy-qhash-namespace]
qhash-namespace/main.cpp:25:1: warning: Move qHash(NS::B *) into NS namespace for ADL lookup [-Wclazy-qhash-namespace]
qhash-namespace/main.cpp:26:1: warning: Move qHash(NS::B::B3) into NS namespace for ADL lookup [-Wclazy-qhash-namespace]
qhash-namespace/main.cpp:30:1: warning: qHash(E) must be declared before QT_END_NAMESPACE [-Wclazy-qhash-namespace]
